From 8a439d20565e3462e0ba6836106c358fa1760a58 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:23 -0400
Subject: [ethernet] igb: Fix ethtool loopback test for 82580 copper

Message-id: <1409062359-13063-6-git-send-email-sassmann@redhat.com>
Patchwork-id: 89125
O-Subject: [RHEL7.1 PATCH 05/81] igb: Fix ethtool loopback test for 82580 copper
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Todd Fujinaka <todd.fujinaka@intel.com>

Add back 82580 loopback tests to ethtool.

Signed-off-by: Todd Fujinaka <todd.fujinaka@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 5aa3a44918db13257ade0bb1caf662f7815de357)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index 48cbc83..86d5142 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -1607,6 +1607,9 @@ static int igb_integrated_phy_loopback(struct igb_adapter *adapter)
    igb_write_phy_reg(hw, I347AT4_PAGE_SELECT, 0);
    igb_write_phy_reg(hw, PHY_CONTROL, 0x4140);
   }
+ } else if (hw->phy.type == e1000_phy_82580) {
+  /* enable MII loopback */
+  igb_write_phy_reg(hw, I82580_PHY_LBK_CTRL, 0x8041);
  }
 
  /* add small delay to avoid loopback test failure */
-- 
1.7.1